storage/client_split_test.go

```diff
@@ -1326,25 +1326,42 @@ func TestStoreRangeSplitRaceUninitializedRHS(t *testing.T) {
 	storeCtx.RaftTickInterval = 50 * time.Millisecond
 	storeCtx.RaftElectionTimeoutTicks = 2
 	currentTrigger := make(chan *roachpb.SplitTrigger, 1)
+	var seen struct {
+		sync.Mutex
+		sids map[storagebase.CmdIDKey][2]bool
+	}
+	seen.sids = make(map[storagebase.CmdIDKey][2]bool)
+
 	storeCtx.TestingKnobs.TestingCommandFilter = func(args storagebase.FilterArgs) *roachpb.Error {
 		et, ok := args.Req.(*roachpb.EndTransactionRequest)
 		if !ok || et.InternalCommitTrigger == nil {
 			return nil
 		}
 		trigger := protoutil.Clone(et.InternalCommitTrigger.GetSplitTrigger()).(*roachpb.SplitTrigger)
+		// The first time the trigger arrives (on each of the two stores),
+		// return a transaction retry. This allows us to pass the trigger to
+		// the goroutine creating faux incoming messages for the yet
+		// nonexistent right-hand-side, giving it a head start. This code looks
+		// fairly complicated since it wants to ensure that the two replicas
+		// don't diverge.
 		if trigger != nil && len(trigger.RightDesc.Replicas) == 2 && args.Hdr.Txn.Epoch == 0 {
```

Comment:
Yes, but if that goes down into this path it will deadlock the test (as it comes with a new command ID).
At the time the test was written there were also Raft reproposals, but we have RaftAppliedIndex now which should prevent those.